src/storageClients/BlobStorageFactsManager.js

```diff
@@ -9,114 +9,114 @@ const FACTS_STORAGE_CONNECTION_STRING = process.env.FACTS_STORAGE_CONNECTION_STR
 
 module.exports = {
 
-    GetFact: function (id, callback, logger) { // eslint-disable-line no-unused-vars
-        var blobSvc = azure.createBlobService(FACTS_STORAGE_CONNECTION_STRING);
-        let blobPrefix = id.split('-')[0] + '/' + id.split('-')[1] + '/' + id.split('-')[2];
-        let factIndex = id.split('-')[3];
-        blobSvc.listBlobsSegmentedWithPrefix(FACTS_CONTAINER_NAME, blobPrefix, null, null, (error, result, response) => { // eslint-disable-line no-unused-vars
-            if (error || !result || !result.entries || result.entries.length == 0) {
-                callback(null, error);
+  GetFact: function (id, callback, logger) { // eslint-disable-line no-unused-vars
+    var blobSvc = azure.createBlobService(FACTS_STORAGE_CONNECTION_STRING);
+    let blobPrefix = id.split('-')[0] + '/' + id.split('-')[1] + '/' + id.split('-')[2];
+    let factIndex = id.split('-')[3];
```

Comment:
I know you didn't change this, but do you know if we want to do something other than throw on a potential invalid index from a malformed id?